### PR TITLE
feat: add create teller component

### PR DESCRIPTION
**Pull request by spider0061**
add the create teller feature.

## Related issues and discussion
#160

## Screenshots, if any
Screenshot from 2020-05-12 19-41-20

## Checklist
Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] If you have multiple commits please combine them into one commit by squashing them.

- [x] Read and understood the contribution guidelines at `web-app/.github/CONTRIBUTING.md`.

### CODE REVIEW

File: src/app/organization/organization-routing.module.ts [hunk 1]
```diff
@@ -163,14 +164,22 @@ const routes: Routes = [
...
+            },
             {
               path: ':id',
               component: ViewTellerComponent,
               data: { title: extract('View Teller'), routeResolveBreadcrumb: ['teller', 'name'] },
               resolve: {
                 teller: TellerResolver
               }
-            }
+            },
```
Comment (Rishabh106) [inline src/app/organization/organization-routing.module.ts]: This change is unnecessary.

File: src/app/organization/tellers/create-teller/create-teller.component.ts [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    const dateFormat = 'yyyy-MM-dd';
+    this.tellerForm.patchValue({
+      startDate: this.datePipe.transform(prevStartDate, dateFormat),
+      endDate: this.datePipe.transform(prevEndDate, dateFormat)
+    });
+    const teller = this.tellerForm.value;
+    teller.locale = 'en';
+    teller.dateFormat = dateFormat;
+    this.organizationService.createTeller(teller).subscribe((response: any) => {
+      this.router.navigate(['../'], { relativeTo: this.route });
```
Comment (Rishabh106) [inline src/app/organization/tellers/create-teller/create-teller.component.ts]: Add resourceId in router navigate to redirect user to 'View Teller' page of the created teller.
```suggestion
      this.router.navigate(['../', response.resourceId], { relativeTo: this.route });
```

File: src/app/organization/organization-routing.module.ts [hunk 3]
```diff
@@ -163,6 +164,14 @@ const routes: Routes = [
                 tellers: TellersResolver
               }
             },
+            {
+              path: 'create',
+              component: CreateTellerComponent,
+              data: { title: extract('Create Teller'), breadcrumb: 'Create Teller' },
```
Comment (karantakalkar) [inline src/app/organization/organization-routing.module.ts]: Hi @spider0061 breadcrumb should be 'Create'  just like other create forms.

File: src/app/organization/tellers/create-teller/create-teller.component.ts [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+  constructor(private formBuilder: FormBuilder,
+              private organizationService: OrganizationService,
+              private route: ActivatedRoute,
+              private router: Router,
+              private datePipe: DatePipe) {
+    this.route.data.subscribe((data: { offices: any }) => {
+      this.officeData = data.offices;
+    });
+    this.tellerStatusesData = [ {'id': 300, 'code': '300', 'value': 'Active'}, {'id': 400, 'code': '400', 'value': 'Inactive'}];
+
```
Comment (karantakalkar) [inline src/app/organization/tellers/create-teller/create-teller.component.ts:48]:
```suggestion

```

File: src/app/organization/tellers/create-teller/create-teller.component.ts [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+   */
+  constructor(private formBuilder: FormBuilder,
+              private organizationService: OrganizationService,
+              private route: ActivatedRoute,
+              private router: Router,
+              private datePipe: DatePipe) {
+    this.route.data.subscribe((data: { offices: any }) => {
+      this.officeData = data.offices;
+    });
+    this.tellerStatusesData = [ {'id': 300, 'code': '300', 'value': 'Active'}, {'id': 400, 'code': '400', 'value': 'Inactive'}];
```
Comment (karantakalkar) [inline src/app/organization/tellers/create-teller/create-teller.component.ts:47]: Line too long, wrap them one below other:
Screenshot from 2020-05-10 11-25-52

File: src/app/organization/tellers/create-teller/create-teller.component.html [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+              </mat-option>
+            </mat-select>
+            <mat-error *ngIf="tellerForm.controls.officeId.hasError('required')">
+              Office is <strong>required</strong>
+            </mat-error>
+          </mat-form-field>
+
+          <mat-form-field>
+            <mat-label>Description</mat-label>
+            <input matInput formControlName="description">
```
Comment (karantakalkar) [inline src/app/organization/tellers/create-teller/create-teller.component.html:36]: this is text area not input search the app for text area formfields

**Comment by spider0061** [comment]
@karantakalkar made the changes. Will remove the MCs once the other PRs gets merged.

**Comment by karantakalkar** [comment]
update screenshot

**Comment by spider0061** [comment]
> update screenshot

done

**Comment by muskankhedia** [comment]
@spider0061 Can you fix the merge conflicts so we can merge the PR.

**Comment by karantakalkar** [comment]
@muskankhedia he'll resolve once we start merging, currently he is very busy with his project.

**Pull request merged**